viamonvif/mdns_server.go

```diff
@@ -16,14 +21,29 @@ type mdnsInfo struct {
 type mdnsServer struct {
 	mu            sync.Mutex
 	mappedDevices map[string]mdnsInfo
+	cacheFilepath string
 	logger        logging.Logger
 }
 
-func newMDNSServer(logger logging.Logger) mdnsServer {
-	return mdnsServer{
+func newMDNSServer(logger logging.Logger) *mdnsServer {
```

Comment:
The change to a pointer was because sync.Mutex objects cannot be copied. I knew when I wrote this in the prior PR I was being too cute leaving `mdnsServer`s return value as a value type. And one patch later it bit me.